hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/Iterators.scala

```diff
@@ -83,8 +94,8 @@ class LogFileIterator(logFiles: List[HoodieLogFile],
     }
     .getOrElse(new TypedProperties())
 
-  protected override val avroSchema: Schema = new Schema.Parser().parse(requiredSchema.avroSchemaStr)
-  protected override val structTypeSchema: StructType = requiredSchema.structTypeSchema
+  protected override val avroSchema: Schema = requiredAvroSchema
```

Comment:
There isn't any functional difference to this code. I just added an alternate constructor so that I didn't have to create another table schema just to deconstruct it
```
  def this(logFiles: List[HoodieLogFile],
            partitionPath: Path,
            tableSchema: HoodieTableSchema,
            requiredSchema: HoodieTableSchema,
            tableState: HoodieTableState,
            config: Configuration) {
    this(logFiles, partitionPath, tableSchema, requiredSchema.structTypeSchema,
      new Schema.Parser().parse(requiredSchema.avroSchemaStr), tableState, config)
  }
```

Reply:
Makes sense.